File: typescript/gigasecond/gigasecond.ts
export default class Gigasecond {
    public static readonly GIGASECOND: number = Math.pow(10, 9) * 1000;
    
    constructor(private readonly startDate: Date) {};

    public date(): Date {
        return new Date(this.startDate.getTime() + Gigasecond.GIGASECOND);
    }
}

File: go/gigasecond/gigasecond.go
package gigasecond

import "time"

const gigasecond time.Duration = time.Duration(1e9 * time.Second)

// AddGigasecond adds 10 ^ 9 seconds to a given date and returns it.
func AddGigasecond(startDate time.Time) time.Time {
	return startDate.Add(gigasecond)
}
